app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,61 @@
+class PostsController < ApplicationController
+
+  before_action :authenticate_user!, :except => [:index, :show]
+
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.create(post_params)
+    @post.update_attribute(:user, current_user)
+    if @post.save == true
+      redirect_to posts_path
+    else
+      render 'new'
+    end
+  end
+
+  def show
+    @post = Post.find(params[:id])
+  end
+
+  def edit
+    @post = Post.find(params[:id])
+    unless current_user.is_post_owner? @post
```

Comment:
Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression. (https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)